### PR TITLE
fix: catalog explore error

### PR DESCRIPTION
**Pull request by rephus**
Closes: https://github.com/lightdash/lightdash/issues/10270

### Description:

[Screencast from 31-05-24 13:53:34.webm](https://github.com/lightdash/lightdash/assets/1983672/5ba7ac22-ec13-4193-8bd2-dc50135dde54)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/catalog/components/CatalogPanel.tsx [hunk 1]
```diff
@@ -200,26 +202,40 @@ export const CatalogPanel: FC = () => {
...
+                    if (item.errors !== undefined) {
+                        if (!acc[TABLES_WITH_ERRORS_GROUP_NAME]) {
+                            acc[TABLES_WITH_ERRORS_GROUP_NAME] = {
+                                name: TABLES_WITH_ERRORS_GROUP_NAME,
+                                tables: {},
+                            };
+                        }
+                        acc[TABLES_WITH_ERRORS_GROUP_NAME].tables[item.name] = {
+                            ...item,
+                            groupName: 'Tables with errors',
```
Comment (joaoviana) [inline packages/frontend/src/features/catalog/components/CatalogPanel.tsx:214]:
```suggestion
                            groupName: TABLES_WITH_ERRORS_GROUP_NAME,
```

File: packages/frontend/src/features/catalog/components/CatalogPanel.tsx [hunk 2]
```diff
@@ -296,6 +312,10 @@ export const CatalogPanel: FC = () => {
                 field: selectedItem.field,
             });
 
+            if (selectedItem.group === TABLES_WITH_ERRORS_GROUP_NAME) {
+                setSidebarOpen(false);
+                return; // no metadata for tables with errors
+            }
```
Comment (joaoviana) [inline packages/frontend/src/features/catalog/components/CatalogPanel.tsx:318]: should we move this to the line before ?
```
if (!isSidebarOpen) {
                setSidebarOpen(true);
            }
```

**Review by joaoviana — APPROVED** [review]
looks great!

File: packages/frontend/src/features/catalog/components/CatalogPanel.tsx (hunk 2 above)
Comment (rephus) [reply packages/frontend/src/features/catalog/components/CatalogPanel.tsx:318]: I see, good idea

**Review by joaoviana — APPROVED** [review]

**Pull request merged**